ViennaAdvantage/Areas/VA012/Models/BankStatementModel.cs

```diff
@@ -1171,9 +1136,12 @@ FROM C_PERIOD
                 {
                     _bankStatementLine.SetC_BPartner_ID(0);
                 }
-                if (_formData[0]._ctrlInvoice > 0)
+                //_formData[0]._ctrlInvoice - Type is changed int to string so according to that check the value using IsNullOrEmpty() method
+                if (!string.IsNullOrEmpty(_formData[0]._ctrlInvoice))
                 {
-                    _bankStatementLine.SetC_Invoice_ID(_formData[0]._ctrlInvoice);
+                    //If Invoice is have then set Invoice from Payment
+                    //_bankStatementLine.SetC_Invoice_ID(_formData[0]._ctrlInvoice);
+                    _bankStatementLine.SetC_Invoice_ID(paymentrecord.GetC_Invoice_ID());
```

Comment:
handle null execption when payement record is null..

Reply:
Done.